client/src/components/AddMovieForm/AddMovieForm.js

```diff
@@ -0,0 +1,112 @@
+import { useState } from 'react'
+import styled from 'styled-components/macro'
+import axios from 'axios'
+import ButtonGreen from '../Buttons/ButtonGreen/ButtonGreen'
+import SuccessCard from '../Messages/SuccessCard/SuccessCard'
+import ErrorCard from '../Messages/ErrorCard/ErrorCard'
+
+const AddMovieForm = () => {
+  const [tmdbId, setTmdbId] = useState('')
+  const [successMessage, setSuccessMessage] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+
+  // Comments will be deleted later on
+  /*
+    The function has to be async, because it has to wait for the response of the tmdb api in the try catch block.
+    Once it has an response, the response can be positive or negative.
+    If the response is positive we allow (allowPostMovie) the movie to be posted to our database.
+    If the response is negative we set an error message.
+  */
+  const addMovieHandler = async event => {
+    event.preventDefault()
+    event.target.reset()
+
+    const newMovie = {
+      tmdb_id: tmdbId,
+    }
+
+    let allowPostMovie = false
+
+    try {
+      const response = await axios.get(
+        `https://api.themoviedb.org/3/movie/${tmdbId}?api_key=7a72c13d262ed45c671611991b0f6086`
+      )
+      if (response.status === 200) {
+        allowPostMovie = true
+      } else {
+        setSuccessMessage('')
+        setErrorMessage('Unknown error connecting to tmdb')
+      }
+    } catch (error) {
+      console.error(error)
+      setSuccessMessage('')
+      setErrorMessage('Movie does not exist on tmdb!')
+    }
+
+    if (allowPostMovie) {
+      axios
+        .post(`http://localhost:4000/api/movies`, newMovie)
```

Comment:
Please set a `proxy` in your `package.json`:
```
"proxy": "http://localhost:4000"
```
That way you only need to write:
```suggestion
        .post(`/api/movies`, newMovie)
```
Which is a bit cleaner and will help with the deployment of your app later on.

Reply:
changed!